src/sentry/api/endpoints/organization_events_stats.py

```diff
@@ -201,6 +215,8 @@ def get(self, request: Request, organization: Organization) -> Response:
                 if referrer in ALLOWED_EVENTS_STATS_REFERRERS.union(METRICS_ENHANCED_REFERRERS)
                 else Referrer.API_ORGANIZATION_EVENT_STATS.value
             )
+            if referrer in SENTRY_BACKEND_REFERRERS:
+                query_source = QuerySource.SENTRY_BACKEND
```

Comment:
Can this logic be moved into `is_frontend_request`? Also, is there no other way to verify that the request came from the backend by inspecting the request like the authentication method?

Reply:
`is_frontend_request` is using authentication but really no matter how we implement it a customer's direct api call can look like a frontend request so this is just a good estimate.

I don't think it's a good idea to move this to is_frontend_request because that's really independent of Snuba. Let me create a method in `Endpoint` or a util 🤔